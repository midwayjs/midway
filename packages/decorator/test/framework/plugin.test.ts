import {
  getClassMetadata,
  CLASS_KEY_CONSTRUCTOR,
  PLUGIN_KEY,
  Plugin,
} from '../../src';

class Test {
  constructor(@Plugin('aaa') aaa: any) {
    // ignore
  }

  @Plugin()
  test: any;

  @Plugin('bbb')
  bbb: any;
}

describe('/test/framework/plugin.test.ts', () => {
  it('plugin decorator should be ok', () => {
    let data = getClassMetadata(CLASS_KEY_CONSTRUCTOR, Test);
    expect(data).toStrictEqual({
      0: {
        key: 'aaa',
        type: PLUGIN_KEY,
      },
    });

    data = getClassMetadata(PLUGIN_KEY, Test);
    expect(data).toStrictEqual([
      { key: 'test', propertyName: 'test' },
      { key: 'bbb', propertyName: 'bbb' },
    ]);
  });
});
