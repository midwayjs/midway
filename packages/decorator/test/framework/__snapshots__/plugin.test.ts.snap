// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/framework/plugin.test.ts plugin decorator should be ok 1`] = `
{
  "bbb": {
    "impl": true,
    "key": "plugin",
    "metadata": {
      "identifier": "bbb",
    },
    "propertyName": "bbb",
  },
  "test": {
    "impl": true,
    "key": "plugin",
    "metadata": {
      "identifier": undefined,
    },
    "propertyName": "test",
  },
}
`;
